corehq/apps/users/middleware.py

```diff
@@ -40,6 +41,9 @@ def process_view(self, request, view_func, view_args, view_kwargs):
             request.domain = view_kwargs['domain']
         if 'org' in view_kwargs:
             request.org = view_kwargs['org']
+        if not (request.user and request.user.is_authenticated):
+            # use this to add in the user if the request is using API key auth
+            HQApiKeyAuthentication().is_authenticated(request)
```

Comment:
Looks like this returns an `HttpUnauthorized` response if authentication fails.  I guess since this call ignores the response, that'll just be silent, and `request.user` won't be set, right?  That should be fine, since the request will fail when the authentication is called for real later.

Reply:
Interesting, but yes that sounds fine